core/src/main/java/org/bitcoinj/core/AddressScript.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2018 John Jegutanis
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.bitcoinj.core;
+
+import org.bitcoinj.script.Script.ScriptType;
+
+import java.io.Serializable;
+
+public interface AddressScript extends Serializable, Cloneable {
+    enum AddressFormat {
+        BASE58,
+        BECH32
+    }
+
+    /** Get the script type of this address */
+    ScriptType getScriptType();
+
+    /** Get the version or header value of this address */
```

Comment:
The equivalent of the address header in Bech32 is the HRP (human readable part). Because of the different types (int vs. String) I think this method should not be part of this interface.

Reply:
Slight correction: Bech32 addresses have a [witness version](https://github.com/Coinomi/bech32/blob/java-ref/ref/java/src/main/java/org/bech32/SegwitAddress.java#L37) that is different from the HRP that is only present in the encoded string.

Reply:
Yes, but as far as I understand the witness version serves and entirely different purpose than the base58 address header (extension of the scripting language vs. distinguishing between networks+usecases). So I think it should not be using the same accessor.

Reply:
Good point, removing it...